pkgs/dart_mcp_server/README.md

```diff
@@ -1,142 +1,167 @@
+button above. 
 
-If you are directly editing your mcp.json file, it should look like this:
+Alternatively, you can configure the server manually. Go to **Cursor -> Settings -> Cursor Settings > Tools & Integrations**, and then click **"Add Custom MCP"** or **"New MCP Server"**
+depending on whether you already have other MCP servers configured. Edit the `.cursor/mcp.json` file in your local project (configuration will only apply to this project) or
+edit the global `~/.cursor/mcp.json` file in your home directory (configuration will apply for
+all projects) to configure the Dart MCP server:
 
 ```json
 {
   "mcpServers": {
-    "dart_mcp": {
+    "dart": {
       "command": "dart",
       "args": [
         "mcp-server",
-        "--experimental-mcp-server",
-        "--force-roots-fallback"
+        "--experimental-mcp-server", // Can be removed for Dart 3.9.0 or later
+        "--force-roots-fallback" // Workaround for a Cursor issue with Roots support
       ]
     }
   }
 }
 ```
 
-Each time you make changes to the server, you'll need to restart the server on
-the MCP configuration page or reload the Cursor window (Developer: Reload Window
-from the Command Palette) to see the changes.
+For more information, see the official Cursor documentation for
+[installing MCP servers](https://docs.cursor.com/context/model-context-protocol#installing-mcp-servers).
+
+### GitHub Copilot in VS Code
+
+<!-- TODO: once the dart.mcpServer setting is not hidden, we may be able
+to provide a deep link to the Dart Extension Settings UI for users to
+enable the server. See docs: https://code.visualstudio.com/docs/configure/settings#_settings-editor.
+This may be preferable to adding the deep link button to VS Code's mcp settings. -->
+
+To configure the Dart MCP server with Copilot or any other AI agent that supports the
+[VS Code MCP API](https://code.visualstudio.com/api/extension-guides/mcp), add the following
+to your VS Code user settings:
+```json
+"dart.mcpServer": true
+```
+
+By adding this setting, the Dart VS Code extension will register the Dart MCP Server
+configuration with VS Code so that you don't have to manually configure the server.
+Copilot will then automatically configure the Dart MCP server on your behalf. This is
+a global setting.
+
+Alternatively, you can click the "Add to VS Code" button below to manually configure the server
+in your VS Code user settings.
+
+[![Add to VS Code](_docs/add_to_vs_code.png)](https://vscode.dev/redirect?url=vscode:mcp/install?%7B%22name%22%3A%22dart%22%2C%22command%22%3A%22dart%22%2C%22args%22%3A%5B%22mcp-server%22%2C%22--experimental-mcp%22%5D%7D)
 
-## Development
+Or, you can manually edit the `.vscode/mcp.json` file in your workspace,
+which will only configure the Dart MCP server for the local workspace:
 
-For local development, use the [MCP Inspector](https://modelcontextprotocol.io/docs/tools/inspector).
+```json
+"servers": {
```

Comment:
I think for Workspace usage, we should still encouarge using the `"dart.mcpServer"` setting, but instead just adding it to Workspace Settings (`.vscode/settings.json`) instead of User settings. That way we have the same benefits of controlling all the flags/conditions.

If we encourage using `mcp.json` and/or the button, users might end up with multiple registrations if they change the User Setting in future (or we change the default).